ocean_provider/validation/requests.py

```diff
@@ -48,6 +78,150 @@ def __init__(
         self._processor = CustomRulesProcessor()
 
 
+class RBACValidator:
+    def __init__(
+        self,
+        request_name=None,
+        request=None,
+        assets: Optional[list] = None,
+        algorithms: Optional[list] = None,
+    ):
+        self._request = request
+        if not request:
+            raise RequestNotFound("Request not found.")
+        action_mapping = {
+            "EncryptRequest": "encryptUrl",
+            "InitializeRequest": "initialize",
+            "DownloadRequest": "access",
+            "ComputeRequest": "compute",
+            "ComputeStartRequest": "compute",
+        }
+        self._action = action_mapping[request_name]
+        self._credentials = {
+            "type": "address",
+            "address": get_provider_wallet().address,
+        }
+        self._component = "provider"
+        self._assets = assets
```

Comment:
I think you can simplify these to one-liners using the python ternary operator: self._assets = assets if assets else []